evrMrmApp/src/drvemCML.cpp

```diff
@@ -357,31 +363,61 @@ MRMCML::getPattern(pattern p, unsigned char *buf, epicsUInt32 blen) const
     blen = std::min(plen, blen);
 
     epicsUInt32 val=0;
-    for(epicsUInt32 i=0; i<blen; i++) {
-        size_t cmlword = (i/mult);
-        size_t cmlbit  = (i%mult);
-
-        size_t cpuword, cpubit;
-        bool first; // first bit in CPU word
-
-        if(mult<32) {
-            first = cmlbit==0;
-            cpuword = cmlword;
-            cpubit = 19 - cmlbit;
-        } else {
-            first = cmlbit==0 || cmlbit==8;
-            cpuword = 2*cmlword + (cmlbit<8 ? 0 : 1);
-            cpubit  = cmlbit<8 ? 7-cmlbit : 31-(cmlbit-8);
-        }
-
-        if(first) {
-            val=shadowPattern[p][cpuword];
-        }
-
-        buf[i]=val>>cpubit;
-        buf[i]&=0x1;
-
+    
+    formFactor form = owner.getFormFactor();
+
+    if (owner.version() >=  MRFVersion(2, 7) && form == formFactor_VME64){
+        for(epicsUInt32 i=0; i<blen; i++) {
+            size_t cmlword = (i/mult);
+            size_t cmlbit  = (i%mult);
+
+            size_t cpuword, cpubit;
+            bool first; // first bit in CPU word
+
+            if(mult<32) {
+                first = cmlbit==0;
+                cpuword = cmlword;
+                cpubit = 19 - cmlbit;
+            } else {
+	        first = cmlbit==0 || cmlbit==32;
+                cpuword = 2*cmlword + (cmlbit<32 ? 1 : 0);
+                cpubit  = cmlbit<32 ? cmlbit : (cmlbit-32);
```

Comment:
Based on the changes in the line `384`, I assume that the line `380` should be adapted as well.

@mdavidsaver can you please comment, or provide info, who might be using this at NSLS and who can test.

Reply:
HI @zioven , I am not sure if I can use the Endianness based check. I haven't tried any VME CPUs with little endian.  I can certainly change the condition to an endianness check, if it is preferred.

I put the condition outside the for loop so that it is not executed for each iteration. I am trying to minimize the time of writing. During the writ time no signals will be generated or the timing is wrong.

Reply:
> Based on the changes in the line `384`, I assume that the line `380` should be adapted as well.
> 
> @mdavidsaver can you please comment, or provide info, who might be using this at NSLS and who can test.

Line 380 affects only the hardware with 20-bit CML words, like EVR-VME-230RF. I don't have one for testing. I would keep it untouched so that existing systems are not affected. If people using the EVR-VME-230RF also found this issue, we can correct it then.